src/main/java/io/fabric8/jenkins/openshiftsync/BuildWatcher.java

```diff
@@ -482,29 +485,26 @@ private static void reconcileRunsAndBuilds() {
 
     for (WorkflowJob job : jobs) {
       BuildConfigProjectProperty property = job.getProperty(BuildConfigProjectProperty.class);
+      BuildConfig buildConfig = property.getBuildConfig();
       if (property == null || StringUtils.isBlank(property.getNamespace()) || StringUtils.isBlank(property.getName())) {
         continue;
       }
-
       logger.info("Checking job " + job.toString() + " runs for BuildConfig " + property.getNamespace() + "/" + property.getName());
-
-      BuildList buildList = getAuthenticatedOpenShiftClient().builds()
-        .inNamespace(property.getNamespace()).withLabel("buildconfig=" + property.getName()).list();
-
+      BuildList buildList = getAuthenticatedOpenShiftClient().builds().inNamespace(property.getNamespace()).withLabel("buildconfig=" + property.getName()).list();
       for (WorkflowRun run : job.getBuilds()) {
         boolean found = false;
         BuildCause cause = run.getCause(BuildCause.class);
         for (Build build : buildList.getItems()) {
-          if (cause != null && cause.getUid().equals(build.getMetadata().getUid())) {
+          String buildUID = build.getMetadata().getUid();
+          if (cause != null && (cause.getUid().equals(buildUID))) {
             found = true;
             break;
           }
         }
         if (!found) {
-          deleteRun(run);
+            deleteRunIfPruneDisabled(run, buildConfig);
```

Comment:
should this be `deleteRunIfPruneEnabled`

Reply:
and I mean from a naming perspective ... I see from the compile error at https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_jenkins-sync-plugin/325/pull-ci-openshift-jenkins-sync-plugin-master-e2e-aws-jenkins/72#0:build-log.txt%3A3929 you at least considered that at some point

Reply:
I changed everything to Enabled. Got a little confused iwth the name while writing this. Thank you Gabe :)